### DISCUSSION TITLE
I need help on JWT authentication

### DISCUSSION THREAD

**Discussion by mdiegoli** [requarks/wiki#3483] [Help / Questions]
**Question**
I need to test a custom jwt authentication module.
In the authentication.js:
...
var opts = {};
opts.jwtFromRequest = ExtractJwt.fromHeader("Authorization")
...

Using postman I send a post to http://localhost:3001/login and in the header I create a new key, 'Authorization', with 'JWT ...token...' as value.

As status of answer I receive 200, but in the body I have the login page...

What I have to do? Where I handle redirect url? I miss some configuration?
Thank you

**Host Info**
OS: Windows 10
Wiki.js version: 2.5.170
Database engine: sqlite

**Comment by mdiegoli** [comment]
No suggestions about how to create a JWT authentication module?

**Comment by NGPixel** [comment]
That's not how authentication works. The login payload is sent via the GraphQL API, not via POST to /login.

What are you trying to achieve exactly?

**Comment by mdiegoli** [comment]
I donìt know graphql... 
what I have to do in a nutshell?
I need to implement a sort of sso between my application and wiki.js, can you help me?
Thanks
